uni/lib/view/home/widgets/main_cards_list.dart

```diff
@@ -52,61 +52,95 @@ class MainCardsListState extends State<MainCardsList> {
+                      });
+                    },
+                    header: createTopBar(context),
+                    children: favoriteCardsFromTypes(
                       widget.favoriteCardTypes,
                       context,
                     ),
-                  ],
-                ),
+                  )
+                : ListView(
+                    children: <Widget>[
+                      createTopBar(context),
+                      ...favoriteCardsFromTypes(
+                        widget.favoriteCardTypes,
+                        context,
+                      ),
+                    ],
+                  ),
+          ),
         ),
+        floatingActionButton: isEditing ? createActionButton(context) : null,
       ),
-      floatingActionButton: isEditing ? createActionButton(context) : null,
     );
   }
 
   Widget createActionButton(BuildContext context) {
     return FloatingActionButton(
-      onPressed: () => showDialog<void>(
-        context: context,
-        builder: (BuildContext context) {
-          return AlertDialog(
-            title: Text(
-              S.of(context).widget_prompt,
-              style: Theme.of(context).textTheme.headlineSmall,
-            ),
-            content: SizedBox(
-              height: 200,
-              width: 100,
-              child: ListView(children: getCardAdders(context)),
-            ),
-            actions: [
-              TextButton(
-                child: Text(
-                  S.of(context).cancel,
-                  style: Theme.of(context).textTheme.bodyMedium,
-                ),
-                onPressed: () => Navigator.pop(context),
+      onPressed: () async {
+        final result = await showDialog<bool>(
+          context: context,
+          builder: (BuildContext context) {
+            return AlertDialog(
+              title: Text(
+                S.of(context).widget_prompt,
+                style: Theme.of(context).textTheme.headlineSmall,
+              ),
+              content: SizedBox(
+                height: 200,
+                width: 100,
+                child: ListView(children: getCardAdders(context)),
               ),
-            ],
-          );
-        },
-      ), //Add FAB functionality here
+              actions: [
+                TextButton(
+                  onPressed: () {
+                    Navigator.of(context).pop(false);
+                  },
+                  child: Text(S.of(context).cancel),
+                ),
+                TextButton(
+                  onPressed: () {
+                    Navigator.of(context).pop(true);
+                  },
+                  child: Text(S.of(context).yes),
+                ),
+              ],
+            );
+          },
+        );
+
+        if (result ?? false) {
+          widget.saveFavoriteCards(widget.favoriteCardTypes);
+          setState(() {
+            isEditing = true;
+          });
+        }
```

Comment:
I don't get this. Do we need to save this after the dialog is closed?

Reply:
It provided a means to persist the editing mode even after the dialog was closed. :)